### ISSUE TITLE
Relay data position

### ISSUE THREAD

**Issue by arosien** [jw3/example-daffodil-debug#32]
We've modeled the current debugger "stack" as the traversal through the schema elements, associating each stack frame to the position of the element in the schema source file. The current _data_ position, however, is relayed as a variable within the stack frame, and variables are assumed to be associated with the stack frame source file (the schema file in this case).

To associate the current data position with the actual data file, we could duplicate the schema stack as another thread, and reference its source as the data file at the given position.

It's not clear if both the schema and data files are open and visible if they would independently "scroll" themselves to the updated position when the stack frames change, but we could test it out.

**Comment by arosien** [comment]
Also, linking to https://marketplace.visualstudio.com/items?itemName=ms-vscode.hexeditor for a possible hex data view.

**Comment by arosien** [comment]
The [scope](https://microsoft.github.io/debug-adapter-protocol/specification#Types_Scope) type has optional source and selected line/column properties, but after adding a separate "data" scope, it didn't seem to have any effect on VS Code--when the "data" scoped variables were selected, the UI didn't change focus to the data file.

**Comment by arosien** [comment]
The position is currently shared as a VS Code variable, but to support https://github.com/jw3/example-daffodil-vscode/issues/45 I think we'd need to send a custom event when the position changes so the extension could manage updating a view onto the data.

**Comment by Shanedell** [comment]
@arosien 

do you know that the variable is and/or how I would access it in the extension?

Also do you have a status on sending a custom event? I have created a view, based off the view you had linked to in https://github.com/jw3/example-daffodil-vscode/issues/45, but it will be blocked from any further updating until it the custom event are being sent.

**Comment by arosien** [comment]
I'm not sure what exactly you need, so I'll try to collect all the bits I know about and what I've implemented so far:

* The current data byte position is returned via DAP as a variable named `bytePos1b`. I imagine this is visible from an extension, or changes may be visible via some listener interface, but I don't really know.
* At the same time, when the current data position changes from within a running parse, a [`loadedSource`](https://microsoft.github.io/debug-adapter-protocol/specification#Events_LoadedSource) event is delivered that says the data dump "source" has changed, which triggers a DAP [`source`](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Source) request, and the backend returns a response containing the (current) content of the data dump. This is my hack to provide a view to the data: generate a data dump view in the backend and deliver it to the frontend as an official "source" of the debugging session. Example content:
```
            ├───────────────────────┤               ├────────┤
87654321  0011 2233 4455 6677 8899 aabb ccdd eeff  0123456789abcdef
00000000: 2a73 7472 6565 743a 5269 6467 6577 6f6f  *street:Ridgewoo
00000010: 6420 4369 7263 6c65 2a63 6974 793a 5761  d␣Circle*city:Wa
00000020: 7368 696e 6774 6f6e 2a73 7461 7465 3a4e  shington*state:N
```
* If the extension wants or needs more fine-grained control or notification, I could send a custom event that would relay the data position so a custom view could be updated.

**Comment by arosien** [comment]
Fixed in 928f078.
